### ISSUE TITLE
How to obtain succes rate and normalized reward plots as given in paper?

### ISSUE THREAD

**Issue by meghbhalerao** [TUMcps/human-robot-gym#13]
Hi, I am trying to reproduce the results in the paper. To do so I run the following command (this is for the pick and place task) - 
`bash human_robot_gym/training/icra_2024_environment_evaluation.sh PP PickAndPlace 500 3000000 1000 8 8000 wandb`
I get the following plots on Wandb under the rollout section - 

<img width="1396" alt="Screenshot 2024-11-28 at 7 53 30 PM" src="https://github.com/user-attachments/assets/60035abe-7dbc-437f-b524-97307c2b059a">
<img width="1396" alt="Screenshot 2024-11-28 at 7 53 48 PM" src="https://github.com/user-attachments/assets/b5622dbc-caa5-4fa3-95a5-bf577c2b796c">
<img width="473" alt="Screenshot 2024-11-28 at 7 54 06 PM" src="https://github.com/user-attachments/assets/58c541a4-585e-46b1-bf75-46cd7dd354c4">

<img width="1403" alt="Screenshot 2024-11-28 at 7 54 21 PM" src="https://github.com/user-attachments/assets/3abde3a8-4538-4ba0-891d-01d20ed4b7ea">

However I am unable to figure out how these translate to the actual plots that are reported in the paper, i.e. normalizd reward and success rate.

Could you provide some details as to how to reproduce these plots?

**Comment by JakobThumm** [comment]
Of course, thank you for that question.
Our plotting pipeline consists of three steps:

1. Training -> Save to tensorboard files
2. Convert tensorboard files to csv files using `utils/data_pipeline.py`
3. Plot csv using `tikz`.

The normalization happens in step 3, therefore it is not part of the official repo. We normalize the data from a range of a manually defined minimum to the measured expert performance.
```
MIN_REWARD = {
    'reach': -50,
    'pp': -1000,
    'cl': 100,
    'rhh': -1000,
    'hrh': -1000,
    'cs': -3000
}
```
I can provide the entire plotting pipeline as well if you need it.

In the case of your pick and place task, the normalization bounds are 
- reward: min = -1000, max = -171.988 (expert)
- success: always just plotted between 0 and 1. Expert performance was 0.944
From what I see in your WandB plots, your trained PP agent is right around the expert performance, which would equal a value of ~1 in our paper plots.

**Comment by meghbhalerao** [comment]
Thanks for your suggestions! I just have a few follow up questions - 

1. Can the final paper plots be done even if I run the training in the wandb mode (and not tensorboard mode?)? I ask this since you mention to save the training in the tensorboard files. I see in the `data_pipeline.py` file that it seems to be using only tensorboard, but just wanted to confirm this once.
2. Thanks for specifying the manual minimum (if this could be included in the docs or readme, that would be great!). Regarding the measured expert performance - where can these numbers be obtained? Could you point me to a file about this? Is it done somewhere here in the `create_expert_dataset.py` file - https://github.com/TUMcps/human-robot-gym/blob/mirror/human_robot_gym/training/create_expert_dataset.py#L49
3. If you could provide the entire training pipeline - maybe something like a shell script + tikz script so that it can save a .pdf  directly from the training data, that would be really great.
Thanks for your help!
4. Is the `rollout/n_goal_reached` in the wandb plots the same as success rate (in the plots in the paper)? 

Thanks for the help!

**Comment by JakobThumm** [comment]
I added our entire results pipeline to the repo. I hope that helps. When running in wandb mode, you should still have the tensorboard files locally.
